### PR TITLE
[minor] Use delimiter argument for `ucwords()`, removing old extra calls (since php >= 5.5.16)

### PR DESCRIPTION
**Pull request by phansys**
| Q | A |
| --- | --- |
| Branch | master |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Container.php [hunk 1]
```diff
@@ -358,7 +358,7 @@ public function getServiceIds()
      */
     public static function camelize($id)
     {
-        return strtr(ucwords(strtr($id, array('_' => ' ', '.' => '_ ', '\\' => '_ '))), array(' ' => ''));
+        return str_replace(['_', '.', '\\', ' '], ['', '_ ', '_ ', ''], ucwords($id, ' _.\\'));
```
Comment (fabpot) [inline src/Symfony/Component/DependencyInjection/Container.php]: We are still using `array()` in Symfony to keep consistency across the board.

**Comment by javiereguiluz** [comment]
@phansys we need to be extra careful about this kind of changes because they usually introduce bugs and bc breaks unawarely.

**Comment by phansys** [comment]
I know @javiereguiluz, fortunately the result of these methods are already covered by tests:
* https://github.com/symfony/symfony/blob/master/src/Symfony/Component/DependencyInjection/Tests/ContainerTest.php#L35
* https://github.com/symfony/symfony/blob/master/src/Symfony/Component/PropertyAccess/Tests/PropertyAccessorTest.php#L87

BTW, given the AppVeyor failure, I've noticed that I misread the current PHP constraint at `composer.json` (`5.5.9`), so I've added a check for the possible scenarios, but adding these checks doesn't seems so happy as if all the supported PHP versions were compatible with this change.

**Comment by fabpot** [comment]
Is it worth the added complexity? Is it faster or slower as there are some more conditions? Does not look worth it to me.

**Comment by phansys** [comment]
I think this doesn't worth, at least until the PHP version will be bumped to something greater than `5.6.16`.
Here is a benchmark with 100000 iterations:
```
Old syntax: 0.21572709083557 s
New syntax: 0.16464805603027 s
```
So, closing now.
Thank you.

**Pull request closed without merging**